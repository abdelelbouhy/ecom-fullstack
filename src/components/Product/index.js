import React from 'react';
import './index.scss';

const Product = (props) => {
    const {title, image: {path, alt}, description, priceLabel, price} = props;

    return (
        <div className='product'>
            <img src={`dist${path}`} alt={alt} className='product_image' />
            <div className='bestseller'>bestseller</div>
            <h1 className='title'>{title}</h1>
            <div className='product_description'>{description}</div>
            <div className='product_price_wrapper'>
                <span className='product_price_wrapper-label'>{priceLabel}</span>
                <span className='product_price_wrapper-price'>{price.toFixed(2)}</span>
            </div>
            <div className='product_shop_now'>Shop Now</div>
        </div>
    );
};


export default Product;