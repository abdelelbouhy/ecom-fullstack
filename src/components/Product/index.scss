.product {
  .product_image {
    width: 100%;
  }
}

@media screen and (min-device-width: 768px) {
  .product {
    width: 33%;
  }
}

@media screen and (min-width: 768px) {
  .product {
    width: 33%;
  }
}

@media screen and (min-device-width: 480px) and (max-device-width: 768px) {
  .product {
    width: 50%;
  }
}

@media screen and (min-width: 480px) and (max-width: 768px) {
  .product {
    width: 50%;
  }
}



@media screen and (min-device-width: 320px) and (max-device-width: 480px) {
  .product {
    width: 100%;
  }
}

@media screen and (min-width: 320px) and (max-width: 480px) {
  .product {
    width: 100%;
  }
}
