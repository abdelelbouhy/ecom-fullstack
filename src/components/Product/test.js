import React from 'react';
import {shallow} from 'enzyme';
import Product from './';


describe('Product Component', () => {
    it('should render', () => {
        const props = {
            title: 'Test Title',
            image : {
                path: '/test.jpg',
                alt: 'Alt'
            },
            description: 'Description',
            priceLabel: 'PriceLabel',
            price: 1200
        };
        const wrapper = shallow(<Product {...props} />);

        expect(wrapper).toMatchSnapshot();
    });
});
