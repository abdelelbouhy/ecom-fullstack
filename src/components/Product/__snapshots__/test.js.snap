// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Component should render 1`] = `
<div
  className="product"
>
  <img
    alt="Alt"
    className="product-product_image"
    src="dist/test.jpg"
  />
  <div
    className="product-bestseller"
  >
    bestseller
  </div>
  <div
    className="product-product_details"
  >
    <h1
      className="product-product_details-title"
    >
      Test Title
    </h1>
    <div
      className="product-product_details-description"
    >
      Description
    </div>
    <div
      className="product-product_details-price_wrapper"
    >
      <span
        className="product-product_details-price_wrapper-label"
      >
        PriceLabel
      </span>
      <span
        className="product-product_details-price_wrapper-price"
      >
        1200.00
      </span>
    </div>
  </div>
  <div
    className="product-shop_now"
  >
    Shop Now
  </div>
</div>
`;
